src/python/WMCore/JobSplitting/EventAwareLumiByWork.py

```diff
@@ -0,0 +1,289 @@
+        collectionName = kwargs.get('collectionName', None)
+        splitOnRun = kwargs.get('splitOnRun', True)
+        getParents = kwargs.get('include_parents', False)
+        runWhitelist = kwargs.get('runWhitelist', [])
+        runs = kwargs.get('runs', None)
+        lumis = kwargs.get('lumis', None)
+        deterministicPileup = kwargs.get('deterministicPileup', False)
+
+        # Set the lumi mask for the fileset based on runs & lumis or runWhitelist or both
+        lumiMask = LumiList()
+        if runs and lumis and runWhitelist:
+            lumiMask = LumiList(wmagentFormat=(runs, lumis)) & LumiList(runs=runWhitelist)
+        elif runs and lumis:
+            lumiMask = LumiList(wmagentFormat=(runs, lumis))
+        elif runWhitelist:
+            lumiMask = LumiList(runs=runWhitelist)
+
+        self.perfParameters = kwargs.get('performance', {})
+        eventsPerLumiInDataset = 0
+
+        # EWV Not looked at yet
+
+        if deterministicPileup and self.package == 'WMCore.WMBS':
+            getJobNumber = self.daoFactory(classname="Jobs.GetNumberOfJobsPerWorkflow")
+            jobNumber = getJobNumber.execute(workflow=self.subscription.getWorkflow().id)
+            self.nJobs = jobNumber
+        # EWV End not looked at yet
+
+        # If we have runLumi info, we need to load it from couch
+        if collectionName:
+            lumiMask = self.lumiMaskFromACDC(couchURL=kwargs.get('couchURL'), couchDB=kwargs.get('couchDB'),
+                                             filesetName=kwargs.get('filesetName'), collectionName=collectionName,
+                                             owner=kwargs.get('owner'), group=kwargs.get('group'), ignoreACDC=ignoreACDC)
+
+        filesByLocation = self.sortByLocation()
+
+        # First we need to load the data
+        if self.package == 'WMCore.WMBS':
+            loadRunLumi = self.daoFactory(classname="Files.GetBulkRunLumi")
+
+        totalJobs = 0
+        totalEvents = 0
+        lastRun = None
+        stopTask = False
+        lastFile = None
+        stopReason = ''
+        failReason = ''
+        for location, filesAtLocation in filesByLocation.iteritems():
+            self.newGroup()  # For each location, we need a new jobGroup
+            failNextJob = False
+            stopJob = False
+            eventsInJob = 0
+            jobLumis = []
+            jobFiles = set()
+            lumisProcessed = set()
+
+            filesByLumi, lumisByFile, eventsByLumi = self.fileLumiMaps(filesAtLocation=filesAtLocation, getParents=getParents, lumiMask=lumiMask)
+            for f in filesAtLocation:
+                lfn = f['lfn']
+                if lfn not in lumisByFile:
+                    continue  # There are no lumis of interest in the file
+                for run, lumi in lumisByFile[lfn].getLumis():
+                    if (run, lumi) in lumisProcessed:
+                        continue  # We already saw this lumi and it got included in an earlier job
```

Comment:
Is this really needed ? You populate this in the same loop, so this skips if you iterate over the same element twice ? Isn't that impossible ?